### ISSUE TITLE
init.sh not working: "Can't find virtualenv -- install"

### ISSUE THREAD

**Issue by thundernixon** [rsms/inter#115]
**Describe the bug**

Initially, when I ran `./init.sh`, it wouldn't work for me, but rather, gave me this error:

```
Setting up virtualenv in 'build/venv'
using pip: /usr/local/bin/pip3 pip 18.1 from /usr/local/lib/python2.7/site-packages/pip (python 2.7)
You are using pip version 9.0.3, however version 19.0.1 is available.
You should consider upgrading via the 'pip install --upgrade pip' command.
./init.sh: Can't find virtualenv -- install through '/usr/local/bin/pip3 install --user virtualenv'
```

**Solution**

This was surprisingly simple for me to solve – I just had to follow pip's advice and run `pip install --upgrade pip`, and then things worked as expected. _However_, I had taken some other troubleshooting steps, so I'm not 100% sure that updating was the only thing that fixed it (for instance, I also briefly made and ran a virtual environment without the init script setting it up for me, but then deleting it later).

**Environment**
- OS: macOS 10.14.2 (18C54)
- Commit: [3323cc4a](https://github.com/rsms/inter/tree/3323cc4a5afc941b039bf1ef3d50fa276e486d77)

**Additional context**
Seems to be working fine now, so maybe someone else will find this solution helpful, or maybe there's something in the init script that could prevent this error from the get-go.

**Comment by rsms** [comment]
Wow. It seems you have a really old version of Python installed on your system, and that python installation is faulty — it appears as `/usr/local/bin/pip3` is a symlink to `/usr/local/lib/python2.7/site-packages/pip` which should only contain Python 2.7 packages, but seem to contain pip for Python 3. This might have been caused by a faulty pip installation (mixed python versions in your system invoked with incorrect environment variables.)

Do you mind trying the following and pasting the output here: (just to be sure this is not some obscure bug in init.sh)

```
which pip3
ls -lF /usr/local/bin/pip3
head -1 /usr/local/bin/pip3
cat /usr/local/lib/python2.7/site-packages/pip/__init__.py
```

As an example, this is what I see on my macOS system: (I have Python 3 installed from Homebrew)

```
$ which pip3
/usr/local/bin/pip3
$ ls -lF /usr/local/bin/pip3
lrwxr-xr-x  1 rasmusandersson  admin  33 Jan 15 12:16 /usr/local/bin/pip3@ -> ../Cellar/python/3.7.2_1/bin/pip3
$ head -1 /usr/local/bin/pip3
#!/usr/local/opt/python/bin/python3.7
$ cat /usr/local/lib/python2.7/site-packages/pip/__init__.py
cat: /usr/local/lib/python2.7/site-packages/pip/__init__.py: No such file or directory
$ cat /usr/local/lib/python3.7/site-packages/pip/__init__.py
__version__ = "18.1"
```

To see if your `pip install --upgrade pip` "workaround" truly fixed it, you can simply remove the build directory `rm -rf build` and run `./init.sh` again — everything it generates ends up in the build directory, so removing it effectively resets things.

If you also updated your environment by sourcing init.sh, then make sure to do the above in a new clean shell session.

**Comment by rsms** [comment]
By the way, init.sh is supposed to stop early if `virtualenv` is not installed. Since there are several ways to install it onto the system, the init.sh script doesn't dare making any assumptions which is why it prints an error message and exists rather than going ahead with installing virtualenv with pip.

**Comment by thundernixon** [comment]
Thanks for such a detailed response!

Ohhh woah, yeah I need to find out if my pip is a symlink to old Python ... I definitely did start installing these things before I had much of an idea of what I was doing.

```
💹 which pip3
/usr/local/bin/pip3

💹 ls -lF /usr/local/bin/pip3
lrwxr-xr-x  1 stephennixon  wheel  31 Mar 31  2018 /usr/local/bin/pip3@ -> ../Cellar/python/3.6.5/bin/pip3

💹 head -1 /usr/local/bin/pip3
#!/usr/local/opt/python/bin/python3.6

💹 cat /usr/local/lib/python2.7/site-packages/pip/__init__.py
__version__ = "19.0.1"
```

(If you have any suggestions of how I might get that more in order, I'm very happy to get pointers, but you don't have to clean up my mess 😄)

When I delete the `/build` folder and run `./init.sh` again, it does seem to once again suggest installing a new version of pip. The output is probably waaay longer than you need, but the relevant parts are at the top and bottom, I think. At the end, I try to `make -j all_var` (which seems to be a valid command?), but get an error.

```
type-repos/google-font-repos/inter 💹 ./init.sh
--------------------------------------------------------------------
Setting up virtualenv in 'build/venv'
using pip: /usr/local/bin/pip3 pip 19.0.1 from /usr/local/lib/python2.7/site-packages/pip (python 2.7)
You are using pip version 9.0.3, however version 19.0.1 is available.
You should consider upgrading via the 'pip install --upgrade pip' command.
You are using pip version 9.0.3, however version 19.0.1 is available.
You should consider upgrading via the 'pip install --upgrade pip' command.
using virtualenv: /usr/local/bin/virtualenv
New python executable in /Users/stephennixon/type-repos/google-font-repos/inter/build/venv/bin/python
Installing setuptools, pip, wheel...done.
build/venv/lib/python -> python2.7
build/venv/bin/pip install -r ./requirements.txt
DEPRECATION: Python 2.7 will reach the end of its life on January 1st, 2020. Please upgrade your Python as Python 2.7 won't be maintained after that date. A future version of pip will drop support for Python 2.7.
Collecting glyphsLib==3.1.4 (from -r ./requirements.txt (line 1))
  Using cached https://files.pythonhosted.org/packages/10/1a/a512712f9c97f9062ddd478bb2272f7c08fa1eba8a70cea2f3bee60d1148/glyphsLib-3.1.4-py2.py3-none-any.whl
Collecting skia-pathops==0.2.0.post2 (from -r ./requirements.txt (line 2))
  Using cached https://files.pythonhosted.org/packages/83/fe/3cd3261d8508c04272f7ffab3849ecb7a2c1e1feaccab3b89710b69d5cc8/skia_pathops-0.2.0.post2-cp27-cp27m-macosx_10_6_intel.macosx_10_9_intel.macosx_10_9_x86_64.macosx_10_10_intel.macosx_10_10_x86_64.whl
Collecting ufo2ft==2.6.0 (from -r ./requirements.txt (line 3))
  Using cached https://files.pythonhosted.org/packages/6a/83/9ec48b90b2eafa06332e3a90f2aaa4e0d146b1451054ce70dcf6d6576719/ufo2ft-2.6.0-py2.py3-none-any.whl
Collecting fontmake==1.8.0 (from -r ./requirements.txt (line 4))
  Using cached https://files.pythonhosted.org/packages/55/76/49e33a84cc196d64a40c7a15cc5b85d7f787fcefe4a96d1dfb484fa92b5c/fontmake-1.8.0-py2.py3-none-any.whl
Collecting fs==2.2.0 (from -r ./requirements.txt (line 5))
  Using cached https://files.pythonhosted.org/packages/32/82/2dcee281d9e04039bbd72e48c42f27c5ef2f37f14880fbd9d9ba5826cf1b/fs-2.2.0-py2.py3-none-any.whl
Collecting numpy==1.15.4 (from -r ./requirements.txt (line 8))
  Using cached https://files.pythonhosted.org/packages/c0/b9/2b485bb32d0b26631f433580d90daad5dea830e6dc5bd18c4f227b1829f7/numpy-1.15.4-cp27-cp27m-macosx_10_6_intel.macosx_10_9_intel.macosx_10_9_x86_64.macosx_10_10_intel.macosx_10_10_x86_64.whl
Collecting scipy==1.2.0 (from -r ./requirements.txt (line 9))
  Using cached https://files.pythonhosted.org/packages/01/4a/686c5b7a7b94ca55f06f8c5992ebd3db0ddb4691ec97d8125f5a5b55eaae/scipy-1.2.0-cp27-cp27m-macosx_10_6_intel.macosx_10_9_intel.macosx_10_9_x86_64.macosx_10_10_intel.macosx_10_10_x86_64.whl
Collecting munkres==1.0.12 (from -r ./requirements.txt (line 10))
  Using cached https://files.pythonhosted.org/packages/ea/f3/21d3c23017ba987e77286c5f7b6334366777919ced67f6bb9d48f5f5936c/munkres-1.0.12-py2.py3-none-any.whl
Collecting fonttools>=3.24.0 (from glyphsLib==3.1.4->-r ./requirements.txt (line 1))
  Using cached https://files.pythonhosted.org/packages/c3/91/8ea90c94b0588de092d57ec98914cd42c052fe97ac3d08dd04af9ee4f182/fonttools-3.36.0-py2.py3-none-any.whl
Collecting defcon>=0.3.0 (from glyphsLib==3.1.4->-r ./requirements.txt (line 1))
  Using cached https://files.pythonhosted.org/packages/59/f1/b55f04cb96239ec77ea712f867af30fdbe546280bad3743a3c3ebe4295e6/defcon-0.6.0-py2.py3-none-any.whl
Collecting compreffor>=0.4.5 (from ufo2ft==2.6.0->-r ./requirements.txt (line 3))
  Using cached https://files.pythonhosted.org/packages/63/3a/922c1d824900f16cef79c560b0e43fb8ed24ffc10502cd08d0e00416eba3/compreffor-0.4.6.post1-cp27-cp27m-macosx_10_6_intel.macosx_10_9_intel.macosx_10_9_x86_64.macosx_10_10_intel.macosx_10_10_x86_64.whl
Collecting booleanOperations>=0.8.2 (from ufo2ft==2.6.0->-r ./requirements.txt (line 3))
  Using cached https://files.pythonhosted.org/packages/3d/67/05961a78573d5d4a710939bb9e0de6b828fd2902635db6ea66c0a54bafc5/booleanOperations-0.8.2-py2.py3-none-any.whl
Collecting cu2qu>=1.6.5 (from ufo2ft==2.6.0->-r ./requirements.txt (line 3))
  Using cached https://files.pythonhosted.org/packages/78/5c/5eb51175e9e4fceb7d5c3e405e1fb575549ce5832a890cd6cb1a18763404/cu2qu-1.6.5-cp27-cp27m-macosx_10_6_intel.macosx_10_9_intel.macosx_10_9_x86_64.macosx_10_10_intel.macosx_10_10_x86_64.whl
Collecting enum34>=1.1.6; python_version < "3.4" (from ufo2ft==2.6.0->-r ./requirements.txt (line 3))
  Using cached https://files.pythonhosted.org/packages/c5/db/e56e6b4bbac7c4a06de1c50de6fe1ef3810018ae11732a50f15f62c7d050/enum34-1.1.6-py2-none-any.whl
Collecting MutatorMath>=2.1.2 (from fontmake==1.8.0->-r ./requirements.txt (line 4))
  Using cached https://files.pythonhosted.org/packages/13/ee/662f841aa324b36f5cd12907e6f695cf831dfdbbc526cda4a2ccc20e812e/MutatorMath-2.1.2-py2.py3-none-any.whl
Collecting backports.os~=0.1; python_version < "3.0" (from fs==2.2.0->-r ./requirements.txt (line 5))
  Using cached https://files.pythonhosted.org/packages/20/c2/193ff30cab747f205535a3a8c24fcc7647dbaf0aa667b7dd580a740b8de4/backports.os-0.1.1-py2-none-any.whl
Collecting typing~=3.6; python_version < "3.6" (from fs==2.2.0->-r ./requirements.txt (line 5))
  Using cached https://files.pythonhosted.org/packages/cc/3e/29f92b7aeda5b078c86d14f550bf85cff809042e3429ace7af6193c3bc9f/typing-3.6.6-py2-none-any.whl
Collecting appdirs~=1.4.3 (from fs==2.2.0->-r ./requirements.txt (line 5))
  Using cached https://files.pythonhosted.org/packages/56/eb/810e700ed1349edde4cbdc1b2a21e28cdf115f9faf263f6bbf8447c1abf3/appdirs-1.4.3-py2.py3-none-any.whl
Requirement already satisfied: setuptools in ./build/venv/lib/python2.7/site-packages (from fs==2.2.0->-r ./requirements.txt (line 5)) (40.6.3)
Collecting pytz (from fs==2.2.0->-r ./requirements.txt (line 5))
  Using cached https://files.pythonhosted.org/packages/61/28/1d3920e4d1d50b19bc5d24398a7cd85cc7b9a75a490570d5a30c57622d34/pytz-2018.9-py2.py3-none-any.whl
Collecting six~=1.10 (from fs==2.2.0->-r ./requirements.txt (line 5))
  Using cached https://files.pythonhosted.org/packages/73/fb/00a976f728d0d1fecfe898238ce23f502a721c0ac0ecfedb80e0d88c64e9/six-1.12.0-py2.py3-none-any.whl
Collecting pyclipper>=1.0.5 (from booleanOperations>=0.8.2->ufo2ft==2.6.0->-r ./requirements.txt (line 3))
  Using cached https://files.pythonhosted.org/packages/4f/ac/44fd766b10ef1c7c059e745d1400d61d26e50e5f5582c5bb3a58267fd423/pyclipper-1.1.0.post1-cp27-cp27m-macosx_10_6_intel.macosx_10_9_intel.macosx_10_9_x86_64.macosx_10_10_intel.macosx_10_10_x86_64.whl
Collecting fontMath>=0.4.8 (from MutatorMath>=2.1.2->fontmake==1.8.0->-r ./requirements.txt (line 4))
  Using cached https://files.pythonhosted.org/packages/9a/66/3cd63d390ff2f38c15e31059507aee90220fc29dd3e2532e18b5bcb5fce2/fontMath-0.4.9-py2.py3-none-any.whl
Collecting future (from backports.os~=0.1; python_version < "3.0"->fs==2.2.0->-r ./requirements.txt (line 5))
Installing collected packages: fonttools, defcon, glyphsLib, skia-pathops, compreffor, pyclipper, booleanOperations, cu2qu, enum34, ufo2ft, fontMath, MutatorMath, fontmake, future, backports.os, typing, appdirs, pytz, six, fs, numpy, scipy, munkres
Successfully installed MutatorMath-2.1.2 appdirs-1.4.3 backports.os-0.1.1 booleanOperations-0.8.2 compreffor-0.4.6.post1 cu2qu-1.6.5 defcon-0.6.0 enum34-1.1.6 fontMath-0.4.9 fontmake-1.8.0 fonttools-3.36.0 fs-2.2.0 future-0.17.1 glyphsLib-3.1.4 munkres-1.0.12 numpy-1.15.4 pyclipper-1.1.0.post1 pytz-2018.9 scipy-1.2.0 six-1.12.0 skia-pathops-0.2.0.post2 typing-3.6.6 ufo2ft-2.6.0
Fetching woff2 from https://github.com/google/woff2.git
Cloning into 'build/deps/woff2'...
remote: Enumerating objects: 1126, done.
remote: Total 1126 (delta 0), reused 0 (delta 0), pack-reused 1126
Receiving objects: 100% (1126/1126), 3.45 MiB | 14.40 MiB/s, done.
Resolving deltas: 100% (678/678), done.
Submodule 'brotli' (https://github.com/google/brotli.git) registered for path 'brotli'
Cloning into '/Users/stephennixon/type-repos/google-font-repos/inter/build/deps/woff2/brotli'...
remote: Enumerating objects: 9, done.
remote: Counting objects: 100% (9/9), done.
remote: Compressing objects: 100% (9/9), done.
remote: Total 5808 (delta 1), reused 1 (delta 0), pack-reused 5799
Receiving objects: 100% (5808/5808), 30.54 MiB | 21.90 MiB/s, done.
Resolving deltas: 100% (3710/3710), done.
Submodule path 'brotli': checked out '533843e3546cd24c8344eaa899c6b0b681c8d222'
Submodule 'research/esaxx' (https://github.com/hillbig/esaxx) registered for path 'brotli/research/esaxx'
Submodule 'research/libdivsufsort' (https://github.com/y-256/libdivsufsort.git) registered for path 'brotli/research/libdivsufsort'
Cloning into '/Users/stephennixon/type-repos/google-font-repos/inter/build/deps/woff2/brotli/research/esaxx'...
remote: Enumerating objects: 32, done.
remote: Counting objects: 100% (32/32), done.
remote: Compressing objects: 100% (20/20), done.
remote: Total 32 (delta 10), reused 32 (delta 10), pack-reused 0
Cloning into '/Users/stephennixon/type-repos/google-font-repos/inter/build/deps/woff2/brotli/research/libdivsufsort'...
remote: Enumerating objects: 210, done.
remote: Total 210 (delta 0), reused 0 (delta 0), pack-reused 210
Receiving objects: 100% (210/210), 98.96 KiB | 3.09 MiB/s, done.
Resolving deltas: 100% (129/129), done.
Submodule path 'brotli/research/esaxx': checked out 'ca7cb332011ec37a8436487f210f396b84bd8273'
Submodule path 'brotli/research/libdivsufsort': checked out '5f60d6f026c30fb4ac296f696b3c8b0eb71bd428'
Note: checking out 'a0d0ed7da27b708c0a4e96ad7a998bddc933c06e'.

You are in 'detached HEAD' state. You can look around, make experimental
changes and commit them, and you can discard any commits you make in this
state without impacting any branches by performing another checkout.

If you want to create a new branch to retain commits you create, you may
do so (now or later) by using -b with the checkout command again. Example:

  git checkout -b <new-branch-name>

HEAD is now at a0d0ed7 Merge pull request #114 from garretrieger/master
Building woff2
rm -f  src/font.o  src/glyph.o  src/normalize.o  src/table_tags.o  src/transform.o  src/woff2_dec.o  src/woff2_enc.o  src/woff2_common.o  src/woff2_out.o  src/variable_length.o  src/woff2_compress.o  src/woff2_decompress.o  src/woff2_info.o woff2_compress woff2_decompress woff2_info
/Library/Developer/CommandLineTools/usr/bin/make -C brotli clean
rm -rf bin libbrotli.a
c++ -fno-omit-frame-pointer -no-canonical-prefixes -DFONT_COMPRESSION_BIN -D __STDC_FORMAT_MACROS -std=c++11 -I./brotli/c/include/ -I./src -I./include -DOS_MACOSX  -c -o src/font.o src/font.cc
c++ -fno-omit-frame-pointer -no-canonical-prefixes -DFONT_COMPRESSION_BIN -D __STDC_FORMAT_MACROS -std=c++11 -I./brotli/c/include/ -I./src -I./include -DOS_MACOSX  -c -o src/glyph.o src/glyph.cc
c++ -fno-omit-frame-pointer -no-canonical-prefixes -DFONT_COMPRESSION_BIN -D __STDC_FORMAT_MACROS -std=c++11 -I./brotli/c/include/ -I./src -I./include -DOS_MACOSX  -c -o src/normalize.o src/normalize.cc
c++ -fno-omit-frame-pointer -no-canonical-prefixes -DFONT_COMPRESSION_BIN -D __STDC_FORMAT_MACROS -std=c++11 -I./brotli/c/include/ -I./src -I./include -DOS_MACOSX  -c -o src/table_tags.o src/table_tags.cc
c++ -fno-omit-frame-pointer -no-canonical-prefixes -DFONT_COMPRESSION_BIN -D __STDC_FORMAT_MACROS -std=c++11 -I./brotli/c/include/ -I./src -I./include -DOS_MACOSX  -c -o src/transform.o src/transform.cc
c++ -fno-omit-frame-pointer -no-canonical-prefixes -DFONT_COMPRESSION_BIN -D __STDC_FORMAT_MACROS -std=c++11 -I./brotli/c/include/ -I./src -I./include -DOS_MACOSX  -c -o src/woff2_dec.o src/woff2_dec.cc
c++ -fno-omit-frame-pointer -no-canonical-prefixes -DFONT_COMPRESSION_BIN -D __STDC_FORMAT_MACROS -std=c++11 -I./brotli/c/include/ -I./src -I./include -DOS_MACOSX  -c -o src/woff2_enc.o src/woff2_enc.cc
c++ -fno-omit-frame-pointer -no-canonical-prefixes -DFONT_COMPRESSION_BIN -D __STDC_FORMAT_MACROS -std=c++11 -I./brotli/c/include/ -I./src -I./include -DOS_MACOSX  -c -o src/woff2_common.o src/woff2_common.cc
c++ -fno-omit-frame-pointer -no-canonical-prefixes -DFONT_COMPRESSION_BIN -D __STDC_FORMAT_MACROS -std=c++11 -I./brotli/c/include/ -I./src -I./include -DOS_MACOSX  -c -o src/woff2_out.o src/woff2_out.cc
c++ -fno-omit-frame-pointer -no-canonical-prefixes -DFONT_COMPRESSION_BIN -D __STDC_FORMAT_MACROS -std=c++11 -I./brotli/c/include/ -I./src -I./include -DOS_MACOSX  -c -o src/variable_length.o src/variable_length.cc
c++ -fno-omit-frame-pointer -no-canonical-prefixes -DFONT_COMPRESSION_BIN -D __STDC_FORMAT_MACROS -std=c++11 -I./brotli/c/include/ -I./src -I./include -DOS_MACOSX  -c -o src/woff2_compress.o src/woff2_compress.cc
c++ -fno-omit-frame-pointer -no-canonical-prefixes -DFONT_COMPRESSION_BIN -D __STDC_FORMAT_MACROS -std=c++11 -I./brotli/c/include/ -I./src -I./include -DOS_MACOSX  -c -o src/woff2_decompress.o src/woff2_decompress.cc
c++ -fno-omit-frame-pointer -no-canonical-prefixes -DFONT_COMPRESSION_BIN -D __STDC_FORMAT_MACROS -std=c++11 -I./brotli/c/include/ -I./src -I./include -DOS_MACOSX  -c -o src/woff2_info.o src/woff2_info.cc
/Library/Developer/CommandLineTools/usr/bin/make -C brotli lib
mkdir -p bin/obj/c/common
mkdir -p bin/obj/c/dec
mkdir -p bin/obj/c/enc
mkdir -p bin/obj/c/tools
mkdir -p bin/tmp
cc -O2  -Ic/include \
        -c c/common/dictionary.c -o bin/obj/c/common/dictionary.o
cc -O2  -Ic/include \
        -c c/common/transform.c -o bin/obj/c/common/transform.o
cc -O2  -Ic/include \
        -c c/dec/bit_reader.c -o bin/obj/c/dec/bit_reader.o
c++ -fno-omit-frame-pointer -no-canonical-prefixes -DFONT_COMPRESSION_BIN -D __STDC_FORMAT_MACROS -std=c++11 -I./brotli/c/include/ -I./src -I./include -DOS_MACOSX  -c -o src/convert_woff2ttf_fuzzer.o src/convert_woff2ttf_fuzzer.cc
cc -O2  -Ic/include \
        -c c/dec/decode.c -o bin/obj/c/dec/decode.o
c++ -fno-omit-frame-pointer -no-canonical-prefixes -DFONT_COMPRESSION_BIN -D __STDC_FORMAT_MACROS -std=c++11 -I./brotli/c/include/ -I./src -I./include -DOS_MACOSX  -c -o src/convert_woff2ttf_fuzzer_new_entry.o src/convert_woff2ttf_fuzzer_new_entry.cc
cc -O2  -Ic/include \
        -c c/dec/huffman.c -o bin/obj/c/dec/huffman.o
cc -O2  -Ic/include \
        -c c/dec/state.c -o bin/obj/c/dec/state.o
cc -O2  -Ic/include \
        -c c/enc/backward_references.c -o bin/obj/c/enc/backward_references.o
cc -O2  -Ic/include \
        -c c/enc/backward_references_hq.c -o bin/obj/c/enc/backward_references_hq.o
cc -O2  -Ic/include \
        -c c/enc/bit_cost.c -o bin/obj/c/enc/bit_cost.o
cc -O2  -Ic/include \
        -c c/enc/block_splitter.c -o bin/obj/c/enc/block_splitter.o
cc -O2  -Ic/include \
        -c c/enc/brotli_bit_stream.c -o bin/obj/c/enc/brotli_bit_stream.o
cc -O2  -Ic/include \
        -c c/enc/cluster.c -o bin/obj/c/enc/cluster.o
cc -O2  -Ic/include \
        -c c/enc/compress_fragment.c -o bin/obj/c/enc/compress_fragment.o
cc -O2  -Ic/include \
        -c c/enc/compress_fragment_two_pass.c -o bin/obj/c/enc/compress_fragment_two_pass.o
cc -O2  -Ic/include \
        -c c/enc/dictionary_hash.c -o bin/obj/c/enc/dictionary_hash.o
cc -O2  -Ic/include \
        -c c/enc/encode.c -o bin/obj/c/enc/encode.o
cc -O2  -Ic/include \
        -c c/enc/encoder_dict.c -o bin/obj/c/enc/encoder_dict.o
cc -O2  -Ic/include \
        -c c/enc/entropy_encode.c -o bin/obj/c/enc/entropy_encode.o
cc -O2  -Ic/include \
        -c c/enc/histogram.c -o bin/obj/c/enc/histogram.o
cc -O2  -Ic/include \
        -c c/enc/literal_cost.c -o bin/obj/c/enc/literal_cost.o
cc -O2  -Ic/include \
        -c c/enc/memory.c -o bin/obj/c/enc/memory.o
cc -O2  -Ic/include \
        -c c/enc/metablock.c -o bin/obj/c/enc/metablock.o
cc -O2  -Ic/include \
        -c c/enc/static_dict.c -o bin/obj/c/enc/static_dict.o
cc -O2  -Ic/include \
        -c c/enc/utf8_util.c -o bin/obj/c/enc/utf8_util.o
rm -f libbrotli.a
ar -crs libbrotli.a bin/obj/c/common/dictionary.o bin/obj/c/common/transform.o bin/obj/c/dec/bit_reader.o bin/obj/c/dec/decode.o bin/obj/c/dec/huffman.o bin/obj/c/dec/state.o bin/obj/c/enc/backward_references.o bin/obj/c/enc/backward_references_hq.o bin/obj/c/enc/bit_cost.o bin/obj/c/enc/block_splitter.o bin/obj/c/enc/brotli_bit_stream.o bin/obj/c/enc/cluster.o bin/obj/c/enc/compress_fragment.o bin/obj/c/enc/compress_fragment_two_pass.o bin/obj/c/enc/dictionary_hash.o bin/obj/c/enc/encode.o bin/obj/c/enc/encoder_dict.o bin/obj/c/enc/entropy_encode.o bin/obj/c/enc/histogram.o bin/obj/c/enc/literal_cost.o bin/obj/c/enc/memory.o bin/obj/c/enc/metablock.o bin/obj/c/enc/static_dict.o bin/obj/c/enc/utf8_util.o
c++   src/font.o  src/glyph.o  src/normalize.o  src/table_tags.o  src/transform.o  src/woff2_dec.o  src/woff2_enc.o  src/woff2_common.o  src/woff2_out.o  src/variable_length.o brotli/bin/obj/c/common/*.o brotli/bin/obj/c/enc/*.o brotli/bin/obj/c/dec/*.o src/woff2_compress.o -o woff2_compress
c++   src/font.o  src/glyph.o  src/normalize.o  src/table_tags.o  src/transform.o  src/woff2_dec.o  src/woff2_enc.o  src/woff2_common.o  src/woff2_out.o  src/variable_length.o brotli/bin/obj/c/common/*.o brotli/bin/obj/c/enc/*.o brotli/bin/obj/c/dec/*.o src/woff2_decompress.o -o woff2_decompress
c++   src/font.o  src/glyph.o  src/normalize.o  src/table_tags.o  src/transform.o  src/woff2_dec.o  src/woff2_enc.o  src/woff2_common.o  src/woff2_out.o  src/variable_length.o brotli/bin/obj/c/common/*.o brotli/bin/obj/c/enc/*.o brotli/bin/obj/c/dec/*.o src/woff2_info.o -o woff2_info
ar cr src/convert_woff2ttf_fuzzer.a  src/font.o  src/glyph.o  src/normalize.o  src/table_tags.o  src/transform.o  src/woff2_dec.o  src/woff2_enc.o  src/woff2_common.o  src/woff2_out.o  src/variable_length.o \
              brotli/bin/obj/c/common/*.o brotli/bin/obj/c/enc/*.o brotli/bin/obj/c/dec/*.o src/convert_woff2ttf_fuzzer.o
ar cr src/convert_woff2ttf_fuzzer_new_entry.a  src/font.o  src/glyph.o  src/normalize.o  src/table_tags.o  src/transform.o  src/woff2_dec.o  src/woff2_enc.o  src/woff2_common.o  src/woff2_out.o  src/variable_length.o \
              brotli/bin/obj/c/common/*.o brotli/bin/obj/c/enc/*.o brotli/bin/obj/c/dec/*.o src/convert_woff2ttf_fuzzer_new_entry.o
build/venv/bin/woff2_compress -> ../../deps/woff2/woff2_compress
Fetching https://github.com/khaledhosny/ots/releases/download/v7.1.7/ots-7.1.7-osx.zip
######################################################################## 100.0%
curl: (6) Could not resolve host: github-production-release-asset-2e65be.s3.amazonaws.com

type-repos/google-font-repos/inter 💹 make -j all_var
--------------------------------------------------------------------
Makefile:75: build/etc/generated.make: No such file or directory
make: *** No rule to make target `build/etc/generated.make'.  Stop.
```

I do have pip3 and pip 3.6 in my /bin that symlink to `/usr/local/Cellar/python/3.6.5/bin/`. However, there is also a pip executable there that includes `#!/usr/local/opt/python@2/bin/python2.7` at the top of its code, so that might be the culprit?  

![image](https://user-images.githubusercontent.com/7355414/51793917-16a16e00-2196-11e9-8526-48ee1321a1a8.png)

![image](https://user-images.githubusercontent.com/7355414/51793922-28831100-2196-11e9-9373-d43aa8143330.png)
 
![image](https://user-images.githubusercontent.com/7355414/51793938-8879b780-2196-11e9-8fcc-4097f70aa6b8.png)


--- 

I definitely love that this allows building with Docker – that's something I'll have to try at some point, to make my own projects more easily buildable by others.

**Comment by rsms** [comment]
> However, there is also a pip executable there that includes `#!/usr/local/opt/python@2/bin/python2.7` at the top of its code, so that might be the culprit?

That's definitely wrong! (assuming that "pip" file is in the Python 3 directory.)

What's happening for you is that somehow "pip3" (specifically `/usr/local/bin/pip3`) really points to Python 2, rather than Python 3 (and `#!/usr/local/opt/python@2/bin/python2.7` at the top of the file `/usr/local/bin/pip3` would cause that to happen.)

It seems these were installed via Homebrew (just a guess based on the path names), and if so, you could try to simply uninstall the `python2` and `python` packages, then reinstall them again.

Something like this:

```
brew uninstall --force python python2 python3
brew cleanup
brew install python
```

You can additionally do `brew install python2` if you also need Python 2 for some reason (e.g. if you are dealing with some older Python code somewhere.)

Alternatively to the above, you could try to manually fix `/usr/local/bin/pip3` to have its "shebang" line edited to point to Python 3.

I'm closing this for now. We can keep talking over email.

**Comment by rsms** [comment]
Glad you got this sorted out!

Perhaps we should fetch & locally install virtualenv in the build directory instead of requiring it to be available from the system...

**Comment by thundernixon** [comment]
As a small update here, I found today that, after all my changing of Python things, I was getting a confusing error when trying to update a dependency in another project:

```
dyld: Library not loaded: @executable_path/../.Python
  Referenced from: /Users/stephennixon/type-repos/google-font-repos/signika-for-google/virtual_env/bin/python3.6
  Reason: image not found
```

I ran `brew doctor` and `brew cleanup`, but the issue persisted.

This was solved by simply `rm -rf`ing the virtual environment in the other project, creating a new one, and installing the dependencies fresh.

I'm not sure whether that relates to your last statement, but I just wanted to put the solution here in case others (or myself, in the future...) come across a similar error.
